contracts/rewards/src/contract/execute.rs

```diff
@@ -42,23 +43,28 @@ where
     /// block height and the epoch duration. If the epoch duration is updated, we store the epoch
     /// in which the update occurs as the last checkpoint
     fn current_epoch(&self, cur_block_height: u64) -> Result<Epoch, ContractError> {
+        assert!(
+            cur_block_height <= i64::MAX as u64,
+            "current block height should fit into a 64bit integer"
+        );
+
         let stored_params = self.store.load_params();
         let epoch_duration: u64 = stored_params.params.epoch_duration.into();
-        let epoch = stored_params.last_updated;
-        if cur_block_height < epoch.block_height_started {
-            return Err(ContractError::BlockHeightInPast.into());
-        }
-        if cur_block_height < epoch.block_height_started + epoch_duration {
-            return Ok(epoch);
+        let last_updated_epoch = stored_params.last_updated;
+
+        let epochs_elapsed = (cur_block_height as i64)
+            .saturating_sub_unsigned(last_updated_epoch.block_height_started)
+            / (epoch_duration as i64); // epoch duration is always positive, so this is safe
```

Comment:
Does this compute to 0 if cur_block_height is in the past, but less than an epoch duration before last_updated_epoch.block_height_started? For example, cur_block_height is 9, last_updated_epoch.block_height_started is 10, and epoch_duration is 5. The subtraction computes -1, and then -1 / 10 = 0. Is that right?

Reply:
damn this stupid calculation!